plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -880,7 +880,7 @@ private static ReaderPageSourceWithRowPositions createParquetPageSource(
 
             List<org.apache.parquet.schema.Type> parquetFields = readBaseColumns.stream()
                     .map(column -> parquetIdToField.get(column.getId()))
-                    .collect(toList());
+                    .toList();
```

Comment:
```suggestion
.collect(toImmutableList());
```

Reply:
This can't use immutable list because this list contains nulls.